app/helpers/application_helper.rb

```diff
@@ -560,4 +560,15 @@ def rss_creator(user)
       end
     end
   end
+
+  def authentication_data
+    @authentication_data ||= begin
```

Comment:
This pattern doesn't memoize `nil` values. Perhaps this is more fitting here?

```suggestion
    return @authentication_data if defined?(@authentication_data)

    @authentication_data = begin
```